assignment1/src/main/java/ro/utcn/sd/he/assignment1/Controller/CommandLineController.java

```diff
@@ -0,0 +1,176 @@
+            case "list":
+                List<Question> questions = questionService.listQuestions();
+                questions.sort(Comparator.comparing(Question::getCreation_date_time).reversed());
+                for(Question q : questions){
+                    print(q.toString());
+                }
+                return false;
+
+            case "login":
+                if(isLogged()){
+                    print("you are already logged in.");
+                } else {
+                    handleLogin();
+                }
+                return false;
+            case"logout":
+                if(isLogged()){
+                    user = null;
+                    print("Logged out");
+                } else {
+                    print("You need to be logged in to log out");
+                }
+                return false;
+            case "add":
+                handleAdd();
+                return false;
+            case "delete":
+                handleDelete();
+                return false;
+            case "exit":
+                return true;
+
+            case "whoami":
+                if(user == null){
+                    print("You are not logged in");
+                } else {
+                    print("Username" + user.getUsername());
+                }
+                return false;
+
+            default:
+                System.out.println("Command unknwon");
+                return false;
+        }
+
+    }
+
+    private void handleSearch(){
+        print("what search criteria you want to use(title/tags):");
+        String criteria = scanner.next().trim();
+        if(criteria.equals("title")){
+            print("Enter title to search:");
+            String title1 = scanner.nextLine();///whyyyyyy
+            String title = scanner.nextLine();///whyyyyyy
+            List<Question> questions = questionService.listQuestions();
+            questions.sort(Comparator.comparing(Question::getCreation_date_time).reversed());;
+            questions.forEach( question -> {
+                if(question.getTitle().indexOf(title) > -1){
+                    print(question.toString());
+                }
+            });
+        }else if (criteria.equals("tags")){
+            print("enter tag:");
+            Integer i = new Integer(0);
+            String tagName = scanner.next().trim();
+            Tag tag = tagService.getTag(tagName);
+            List<Question> questions =  questionTagService.getQuestionsWithTag(tag);
+            if(questions.isEmpty()){
+                print("No questions found with the given tag");
+            }else{
+                questions.forEach(question -> print(question.toString()));
+            }
+
+        } else {
+            print("not known criteria");
+        }
+    }
+
+    private void handleAdd(){
+        if(isLogged()){
+            print("enter the title:");
+            String title = scanner.nextLine();
+            title = scanner.nextLine(); ////whyyyyyyy
```

Comment:
You do not need to assign the return value of the scanner.nextLine() if you want to discard it (basically the first assignment is redundant). Also, even though I understand why you included the "////whyyyyyyy" comment, remove it :P And maybe explain that you want to just discard an empty line.